Cargo.toml

```diff
@@ -6,3 +6,6 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+color-eyre = "0.6.2"
+eyre = "0.6.8"
+thiserror = "1.0.35"
```

Comment:
`color-eyre` is a crate to neatly display errors via `color_eyre::Report.`

technically the `eyre` here is not needed, but included for completeness.

`thiserror` is a crate to quickly and easily make custom errors. 

`thiserror` is good for libraries, `eyre` and the equivalent `anyhow` is good for binaries.